src/handlers/dynamic_report/get_dynamic_report_handler.py

```diff
@@ -43,40 +41,29 @@ def get_metric_report_service(calculator, profile_range, company_anonymization):
     )
 
 
-def get_investment_year_report_service(report):
-    repository = InvestmentRepository(session, QuerySQLBuilder(), ResponseSQL(), logger)
-
-    return InvestmentYearReport(logger, report, repository)
-
-
-def get_calendar_year_report_service(report):
-    repository = CalculatorRepository(session, QuerySQLBuilder(), ResponseSQL(), logger)
-
-    return ComparisonvsPeersService(logger, report, repository)
-
-
 def get_dynamic_report_service():
     user_service = get_user_details_service_instance()
     company_anonymization = CompanyAnonymization(user_service)
     calculator = CalculatorService(logger)
     profile_range = ProfileRange(session, QuerySQLBuilder(), logger, ResponseSQL())
-    report = CalculatorReport(logger, calculator, profile_range, company_anonymization)
-    calculator_repository = CalculatorRepository(
-        session, QuerySQLBuilder(), ResponseSQL(), logger
+    repository = BaseMetricsRepository(
+        logger, session, QuerySQLBuilder(), ResponseSQL()
+    )
+    base_metric_report = BaseMetricsReport(
+        logger, calculator, profile_range, company_anonymization
     )
+    report = ByYearReportService(logger, base_metric_report, repository)
 
     metric_report = get_metric_report_service(
         calculator, profile_range, company_anonymization
     )
-    investment_report = get_investment_year_report_service(report)
-    calendar_report = get_calendar_year_report_service(report)
+    calendar_report = ByYearReportService(logger, report, repository)
```

Comment:
calendar_report and report(line 55) are using the same class ByYearReportService and it's returning this error message `"error": "\'ByYearReportService\' object has no attribute \'set_company_permissions\'"`. I think line 60 should be change to `calendar_report = ByYearReportService(logger,  base_metric_report, repository)` and delete line 55